nemo/collections/asr/parts/mixins/mixins.py

```diff
@@ -76,13 +76,12 @@ def _setup_tokenizer(self, tokenizer_cfg: DictConfig):
 
             if 'special_tokens' in self.tokenizer_cfg:
                 special_tokens = self.tokenizer_cfg['special_tokens']
-            else:
-                special_tokens = None
+
+                if special_tokens is not None:
+                    raise ValueError("`special_tokens` are no longer supported for SentencePiece based tokenizers.")
 
             # Update special tokens
-            self.tokenizer = tokenizers.SentencePieceTokenizer(
-                model_path=model_path, special_tokens=special_tokens, legacy=True
-            )
+            self.tokenizer = tokenizers.SentencePieceTokenizer(model_path=model_path)
```

Comment:
What there a reason "legacy=True" is dropped?

Reply:
its not necessary any more. Now we use script to embed the EOS, BOS and PAD tokens along with UNK. Previously this was handled in a weird nemo specific manner (legacy mode). 
Since we are supporting it embedded in the tokenizer now, its good to drop legacy support too.